### PR TITLE
refactoring change company colour scheme args

### PR DESCRIPTION
**Pull request by reinaldorauch**

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/GameCommands.h [hunk 1]
```diff
@@ -696,28 +696,56 @@ namespace OpenLoco::GameCommands
...
+            , colourType(regs.cl)
+            , setColourMode(regs.dh)
+        {
+            if (setColourMode == 0)
+            {
+                isPrimary = regs.ah == 0;
+            }
+            else if (setColourMode == 1)
+            {
+                isPrimary = regs.al;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/GameCommands.h:717]: isPrimary is only used for setColourMode 0. Remove this line.

Comment (reinaldorauch) [reply src/OpenLoco/src/GameCommands/GameCommands.h:717]: Will do

File: src/OpenLoco/src/Windows/CompanyWindow.cpp [hunk 2]
```diff
@@ -1418,7 +1418,15 @@ namespace OpenLoco::Ui::Windows::CompanyWindow
 
                     GameCommands::setErrorTitle(StringIds::error_cant_change_colour_scheme);
 
-                    GameCommands::do_19(0, newMode, vehicleType, 1, CompanyId(self.number));
+                    GameCommands::ChangeCompanyColourSchemeArgs args{};
+
+                    args.isPrimary = 0;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/CompanyWindow.cpp:1423]: false

File: src/OpenLoco/src/Windows/CompanyWindow.cpp [hunk 3]
```diff
@@ -1527,7 +1535,16 @@ namespace OpenLoco::Ui::Windows::CompanyWindow
                     const int8_t colour = Dropdown::getItemArgument(itemIndex, 2);
                     const auto vehicleType = widgetIndex - widx::main_colour_scheme;
 
-                    GameCommands::do_19(0, colour, vehicleType, 0, CompanyId(self.number));
+                    GameCommands::ChangeCompanyColourSchemeArgs args{};
+
+                    args.isPrimary = 0;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/CompanyWindow.cpp:1540]: false

File: src/OpenLoco/src/Windows/CompanyWindow.cpp [hunk 4]
```diff
@@ -1551,7 +1568,15 @@ namespace OpenLoco::Ui::Windows::CompanyWindow
                     const int8_t colour = Dropdown::getItemArgument(itemIndex, 2);
                     const auto vehicleType = widgetIndex - widx::secondary_colour_scheme;
 
-                    GameCommands::do_19(1, colour, vehicleType, 0, CompanyId(self.number));
+                    GameCommands::ChangeCompanyColourSchemeArgs args{};
+
+                    args.isPrimary = 1;
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Windows/CompanyWindow.cpp:1573]: true

**Comment by duncanspumpkin** [comment]
Could you also use these args in the `changeCompanyColour` function.

File: src/OpenLoco/src/GameCommands/ChangeCompanyColour.cpp [hunk 5]
```diff
@@ -12,36 +12,36 @@ using namespace OpenLoco::Interop;
 namespace OpenLoco::GameCommands
 {
     // 0x0043483D
-    static uint32_t changeCompanyColour(uint8_t flags, uint8_t type, bool toggleMode, CompanyId companyId, bool isSecondary, uint8_t value)
+    static uint32_t changeCompanyColour(ChangeCompanyColourSchemeArgs args, uint8_t flags)
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/ChangeCompanyColour.cpp:15]: const &

**Review by duncanspumpkin — APPROVED** [review]

File: src/OpenLoco/src/GameCommands/ChangeCompanyColour.cpp (hunk 5 above)
Comment (AaronVanGeffen) [reply src/OpenLoco/src/GameCommands/ChangeCompanyColour.cpp:15]:
```suggestion
    static uint32_t changeCompanyColour(const ChangeCompanyColourSchemeArgs& args, uint8_t flags)
```

**Pull request merged**